src/DependencyInjection/MercureExtension.php

```diff
@@ -67,9 +73,29 @@ public function load(array $configs, ContainerBuilder $container)
             $bus = $hub['bus'] ?? null;
             $attributes = null === $bus ? [] : ['bus' => $hub['bus']];
             $publisherDefinition->addTag('messenger.message_handler', $attributes);
+
+            if ($config['enable_profiler'] && class_exists(Stopwatch::class)) {
+                $container->register("$hubId.traceable", TraceablePublisher::class)
+                    ->setDecoratedService($hubId)
+                    ->setArguments([new Reference("$hubId.traceable.inner"), new Reference('debug.stopwatch')]);
+                $publishers[$name] = new Reference("$hubId.traceable");
+                if ($defaultHub === $hubId) {
+                    $container->setAlias(TraceablePublisher::class, "$hubId.traceable");
+                }
+            }
+        }
+
+        if ($config['enable_profiler']) {
+            $container->register('data_collector.mercure', MercureDataCollector::class)
+                ->setArguments([$publishers, $hubUrls])
```

Comment:
Maybe that a map `hubUrl[publisher]` would be better and less error prone? Or the hub URL could be injected in `TraceablePublisher`'s constructor and then retrieved using a getter?

Reply:
I didn't want to inject the hub Url in the TraceablePublisher because the url is only required by the WebProfiler view on this bundle, not on the lib.

Reply:
Otherwise, we could remove the `$hubUrls` argument, as it's only used to display the hub url on the profiler. I don't find it so useful cause most of the time the user already know the hub url cause he configured it ^^

Reply:
Let's keep it as is, in case updates are sent to several hubs during the same request.

Reply:
Actually it could be useful for other use cases too to be able to access the hub URL. IMHO it fits well in the lib too.